src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -303,10 +307,23 @@ function getInteractiveRouterNavigationCallbacks(): NavigationCallbacks | undefi
   return navigationCallbacks.get(interactiveRouterRendererId);
 }
 
-function shouldUseClientSideRouting() {
-  return hasInteractiveRouter() || !hasProgrammaticEnhancedNavigationHandler();
+function currentPageLoadMechanism(): PageLoadMechanism {
+  if (hasInteractiveRouter()) {
+    return 'clientside-router';
+  } else if (hasProgrammaticEnhancedNavigationHandler()) {
+    return 'serverside-enhanced';
+  } else {
+    // For back-compat, in blazor.server.js or blazor.webassembly.js, we always behave as if there's an interactive
+    // router even if there isn't one attached. This preserves a niche case where people may call Blazor.navigateTo
+    // without a router and expect to receive a notification on the .NET side but no page load occurs.
+    // In blazor.web.js, we explicitly recognize the case where you have neither an interactive nor enhanced SSR router
+    // attached, and then handle Blazor.navigateTo by doing a full page load because that's more useful (issue #51636).
+    return Blazor._internal.isBlazorWeb ? 'serverside-fullpageload' : 'clientside-router';
+  }
```

Comment:
Previously we just had a boolean condition "shouldUseClientSideRouting". However that was misleading because there are really three cases (client-side routing, enhanced nav, and neither). By changing this boolean condition to one with all three possibilities, it forces other code to remember and account for the "neither" case.